Add chains to gitops WIP DNM

simonbaird opened the pull request:
### ---> THIS IS NOW REPLACED BY https://github.com/redhat-appstudio/infra-deployments/pull/111 <---

Old description:

It's kinda working now, but still WIP.

Soon I'll move it into a PR on the main repo.

Todo:
- Deal with multiple records showing up in verify-att.out - DONE
- Rename crc.crt to self-signed-cluster-ingress.pem or similar as suggested - DONE
- Maybe include the S2I demo - DONE

JIRA: HACBS-38
JIRA: HACBS-47

simonbaird committed:
Add gitops config & key-pair script for chains

The gitops config sits under the build component which I think makes
sense, since it's labelled as "part-of" tekton-pipelines.

The script will generate a signing secret needed for chains signing.
You have to run it manually for now.

Also:
- Include one `oc adm policy...` command, which is required to get
  chains working, to the bootstrap-cluster script.
- Set the project at the end of the bootstrap-cluster script. Maybe
  it doesn't matter but it seems reasonable to choose a project
  explicitly rather than let it land wherever it lands, which would
  now be tekton-chains, but previously would have been
  openshift-pipelines.

JIRA: HACBS-38

simonbaird committed:
Two scripts to help get SSL working for chains

1. Add CA certs accessible in the chains controller to prevent TLS
failures when running chains in a local cluster

A workaround based on https://docs.engineering.redhat.com/x/zFp-Dw .
There's probably better ways to solve the problem, but this gets
things working so I think it's okay for now until we have a better
solution.

2. Trust local cluster ingress cert to allow cosign to access the
local registry if it's using a self-signed cert

It might be possible to do the docker login without SSL and then run
cosign with --allow-insecure-registry, but it's easier to avoid all
that by trusting the local cluster's ingress cert.

This is probably not needed or wanted for non-local clusters that
already have real trusted certs, but it should be a useful
convenience for local development using CRC.

JIRA: HACBS-47

Michkov commented on hack/chains/controller-certs-hack.sh:
```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
```
Maybe using https://cert-manager.io/ operator could be nicer.

Michkov commented on hack/chains/controller-certs-hack.sh:
```diff
@@ -0,0 +1,68 @@
+#
+
+# A cert from your local cluster to get local SSL working
+kubectl get secret router-ca -n openshift-ingress-operator -o jsonpath="{.data.tls\.crt}" | base64 -d > $TMP_DIR/crc.pem
```
this is universal, not related to crc, crc name leads to confusion.

simonbaird replied:
In the latest revision I called it "cluster-ingress" instead of "crc-cert"

simonbaird commented on hack/bootstrap-cluster.sh:
```diff
@@ -46,6 +46,11 @@ kubectl apply -f $ROOT/argo-cd-apps/app-of-apps/all-applications-staging.yaml
+echo
+echo "Configuring service account for Tekton Chains:"
+oc new-project tekton-chains
+oc adm policy add-scc-to-user anyuid -z tekton-chains-controller
```
I was wondering if this could be done with a yaml file via the GitOps component rather than doing it here, but I couldn't figure out how.

Michkov replied:
is that needed? I was able run it without this command, but unfortunatelly it was month ago, so do not remember details.

simonbaird replied:
Yeah, it is needed afaict. Without it I can see errors in the controller startup in my local CRC cluster.

simonbaird committed:
Scripts to help demo chains, cosign and rekor-cli

The goal is to make it easy to configure and demonstrate working
chains functionality in an infra-deployments managed cluster.

Based on Christophe's guide at
https://docs.engineering.redhat.com/x/zFp-Dw which is based on the
official chains signed provenance tutorial.

Also includes a readme and some minor improvements and commentary in
the old check-taskrun scripts.

JIRA: HACBS-47

simonbaird commented:
This is now superseded by https://github.com/redhat-appstudio/infra-deployments/pull/111 .

To prevent confusion, it's better to put any further review comments on the new PR.

I'll close this one soon.

simonbaird committed:
Change chains security context config

Remove the `oc adm policy add-scc-to-user ` command and instead use
configuration change so it's not required.

(As suggested by @Michkov in review.)
